lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -199,6 +197,17 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
 
   // Replace the RefResolveOp with verbatim op representing the XMR.
   LogicalResult handleRefResolve(RefResolveOp resolve) {
+    auto resWidth = getBitWidth(resolve.getType());
+    if (resWidth.has_value() && resWidth.value() == 0) {
+      // Donot emit 0 width XMRs, replace it with constant 0.
+      ImplicitLocOpBuilder builder(resolve.getLoc(), resolve);
+      auto zeroUintType = UIntType::get(builder.getContext(), 0);
+      auto zeroC = builder.createOrFold<BitCastOp>(
+          resolve.getType(), builder.create<ConstantOp>(
+                                 zeroUintType, getIntZerosAttr(zeroUintType)));
+      resolve.getResult().replaceAllUsesWith(zeroC);
+      return success();
+    }
```

Comment:
The only further improvement I can think of is to cache the creation of the zero-width constant on a per-module basis. Otherwise, this looks solid.

Reply:
Keeping this open, for a following cleanup.